tools/sync-commits-with-smithy-rs/src/main.rs

```diff
@@ -0,0 +1,318 @@
+use anyhow::{bail, Context, Result};
+use git2::{Commit, Direction, IndexAddOption, ObjectType, Oid, Repository, Signature, Sort};
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "sync")]
+struct Opt {
+    #[structopt(long, parse(from_os_str))]
+    from_repo_at_path: PathBuf,
+    #[structopt(long, parse(from_os_str))]
+    to_repo_at_path: PathBuf,
+}
+
+const BOT_NAME: &str = "AWS SDK Rust Bot";
+const BOT_EMAIL: &str = "aws-sdk-rust-primary@amazon.com";
+const NEXT_BRANCH: &str = "next";
+
+/*
+cargo run -- \
+--from-repo-at-path /Users/zhessler/Documents/smithy-rs/ \
+--to-repo-at-path /Users/zhessler/Documents/aws-sdk-rust/
+
+when aws-sdk-rust/GENERATED_FROM_THIS_SMITHY_RS_COMMIT
+contains 35ac555e0f2c4245b010944290d139afe6753b23
+
+returns:
+    commits:
+    [
+        e38c531464dd4b84b2572fb2831dc0cc95101d01
+        a536c15b6e43a424662dd92ff388316ab735d9a5
+        bc316a0b81b75a00c389f6281a66eb0f5357172a
+        166ffd62f354474af20c5ca6b4737a9ae2cda351
+        17818ca7e5221ff1a82356a9b6259f852dc8cb16
+        341194e2f91437070a16ceac008e3a85f5aee34f
+        2e7ed943513203f1472f2490866dc4fb8a392bd3
+        af1832edf62783ba694c8c9f867e6c1c16b3f444
+        6325c8a277225e786920d42df935371d27b11fc4
+        af6ec39e9f1207e769bdfd34bfdd35fe92fef229
+        5ece12393be6c635265a85416efb34534297dfde
+    ]
+ */
+
+fn main() {
+    let Opt {
+        from_repo_at_path,
+        to_repo_at_path,
+    } = Opt::from_args();
+
+    sync_with_latest(&from_repo_at_path, &to_repo_at_path).unwrap();
+}
+
+fn sync_with_latest(from_repo_path: &Path, to_repo_path: &Path) -> Result<()> {
+    let context = "couldn't sync commits";
+
+    // Check repo that we're going to be moving the code into to see what commit it was last synced with
+    let last_synced_commit = get_last_synced_commit(to_repo_path).unwrap();
+    let commit_revs = commits_to_be_applied(from_repo_path, &last_synced_commit)
+        .context(context)
+        .unwrap();
```

Comment:
did you mean to unwrap here?

Reply:
I assume no? since the function returns result

Reply:
Yeah, that's a leftover from when I was testing a failing thing